src/_pytest/config/__init__.py

```diff
@@ -112,6 +112,17 @@ def directory_arg(path, optname):
     return path
 
 
+# Plugins that cannot be disabled via "-p no:X" currently.
+essential_plugins = (
+    "mark",
+    "main",
+    "terminal",  # Required to display/handle failures.
+    "runner",
+    "python",
+    "fixtures",
+    "helpconfig",  # Provides -p.
+)
+
 default_plugins = (
```

Comment:
I think we should remove explicitly listing `essential_plugins` here, mostly for consistency, and then use `essential_plugins` to initialize `default_plugins`

```
default_plugins = essential_plugins + (
    "debugging",
    "unittest",
    ...
```

Reply:
Like this?
```
diff --git i/src/_pytest/config/__init__.py w/src/_pytest/config/__init__.py
index 2b3faa2e..253c61f0 100644
--- i/src/_pytest/config/__init__.py
+++ w/src/_pytest/config/__init__.py
@@ -123,7 +123,7 @@ def directory_arg(path, optname):
     "helpconfig",  # Provides -p.
 )

-default_plugins = (
+default_plugins = essential_plugins + (
     "mark",
     "main",
     "terminal",
@@ -153,7 +153,7 @@ def directory_arg(path, optname):
     "logging",
 )

-builtin_plugins = set(essential_plugins + default_plugins)
+builtin_plugins = set(default_plugins)
 builtin_plugins.add("pytester")


```

Reply:
Yes, while also removing the duplicated names from `default_plugins` which are already in `essential_plugins`

Reply:
Oh, sure - have re-added that accidentally when comparing the list again - order there appears to matter (at least for some other test in the file), and I've used the order from the test at first.